### DISCUSSION TITLE
Can finbuckle handle multi tenant with hierarchical  structure

### DISCUSSION THREAD

**Discussion by minorityuk** [Finbuckle/Finbuckle.MultiTenant#618] [Q&A]
Hi

I am looking to find a solution to a design problem I have.  A product I have inherited was designed as a database per company asset.

To complicate things our product is provided to service company who then use it for their customer. Where their customer gave multiple assets sitting in individual database with their own authentication.

Ideally what I want to achieve is
- service company can search across all items of their customers and each of their assets.
- their customer gave a admin role who can see all their assets.
- each asset have users who can edit, add to that asset but have read only to other assets.

Currently I am using claims permission to allow add, edit delete based on roles.

Ideally I need to get to level where a user is granted different roles for different assets can this library achieve the above?

**Comment by AndrewTriesToCode** [comment]
Hi, there is no support for this built in, but it should be possible. If you look at the `MultiTenantDbContext` code you should see where the logic for applying the tenant filter is -- you could easily modify that to also include a tenant's "servicer" tenant if that tenant was added as a property on the tenant.
